lib/main.dart

```diff
@@ -81,17 +76,40 @@ class _MyHomePageState extends State<MyHomePage> {
           crossAxisAlignment: CrossAxisAlignment.start,
           children: [
             Expanded(
-              child: ListView.builder(
-                itemCount: messages.length,
-                reverse: true,
-                padding: const EdgeInsets.symmetric(vertical: 8.0),
-                itemBuilder: (context, index) {
-                  return ListTile(
-                    leading: DefaultTextStyle.merge(
-                      style: const TextStyle(color: Colors.indigo),
-                      child: Text(messages[index].timestamp),
-                    ),
-                    title: Text(messages[index].message),
+              child: StreamBuilder<QuerySnapshot>(
+                stream: FirebaseFirestore.instance
```

Comment:
Easy mistake even Google does often ;)
This causes useless re-renderings because every time the `build()` is called, It recreates the stream. So, every time, the `setState` is called, it will also re-render the list. Define the stream as a variable.

Reply:
how can I do that? It won't be fixed by making it as var!

Reply:
hmm, maybe another widget class

Reply:
It's OK to define it as a member variable in `_MyHomePageState` class because this will not be re-constructed for rendering. Rendering only causes calling the `build` function

Reply:
But another class and another file is better.

Reply:
@aliyazdi75 Of course. If we have a controller concept or state management in this app, that would be ideal but let's keep it simple for now.